Hero.cpp

```diff
@@ -0,0 +1,67 @@
+#include "Hero.h"
+
+Hero::Hero(
+		const std::string _name,
+		float _maxHP,
+		float _DMG,
+		float _ACD,
+		float _XPToLevelup,
+		float _maxHPBonus,
+		float _DMGBonus,
+		float _ACDBonus)
+	: Character(_name, _maxHP, _DMG, _ACD)
+{
+	XP = 0;
+	maxHPBonus = _maxHPBonus;
+	DMGBonus = _DMGBonus;
+	ACDBonus = _ACDBonus;
+	XPToLevelup = _XPToLevelup;
+}
+
+Hero Hero::parse(std::string fileName)
+{
+	std::map<std::string, std::any> characterAttributes = JSON::parse(fileName, true);
+	JSON::ruleOutNegativeAnyFloat(characterAttributes["base_health_points"]);
+	JSON::ruleOutNegativeAnyFloat(characterAttributes["base_damage"]);
+	JSON::ruleOutNegativeAnyFloat(characterAttributes["base_attack_cooldown"]);
+	JSON::ruleOutNegativeAnyFloat(characterAttributes["experience_per_level"]);
+	JSON::ruleOutNegativeAnyFloat(characterAttributes["health_point_bonus_per_level"]);
+	JSON::ruleOutNegativeAnyFloat(characterAttributes["damage_bonus_per_level"]);
+	JSON::ruleOutNegativeAnyFloat(characterAttributes["cooldown_multiplier_per_level"]);
+	return Hero(
+        std::any_cast<std::string>(characterAttributes["name"]),
+        std::any_cast<float>(characterAttributes["base_health_points"]),
+        std::any_cast<float>(characterAttributes["base_damage"]),
+        std::any_cast<float>(characterAttributes["base_attack_cooldown"]),
+		std::any_cast<float>(characterAttributes["experience_per_level"]),
+		std::any_cast<float>(characterAttributes["health_point_bonus_per_level"]),
+		std::any_cast<float>(characterAttributes["damage_bonus_per_level"]),
+		std::any_cast<float>(characterAttributes["cooldown_multiplier_per_level"]));
```

Comment:
it miert nem hasznalod a get-et? azert lett csinalva. ugyanez monsternel., hogy végre legyen virtuális futás, küldöm a selfie-t a montázshoz ￼

Reply:
Ez még a korai main.cpp hozománya, amiben nem volt a JSON-nak saját map tagváltozója, nem kellett getterezni.

Reply:
@hegyhati kész vagyok a módosításokkal.   
   
Emellett a documentation PR discussionben lenne pár kérdés:  
1, Jó -e úgy, hogy a documentation ebből az ágból lett indítva, de egyből a masterre nyitottam PR-t? Amíg ez nem megy masterre, addig azt Draft-ként hagyom. Csak szerettem volna, hogy elinduljon a review ott is.   
2, Az osztályoknak elég a public tagjaihoz írni doksit?  
3, Most úgy van beállítva a documentation deploying workflow, hogy PR nyitáskor, illetve nyitott PR branchére commitoláskor fusson. Erre mi lenne jobb megoldás? Mindig a mastert mutassa be? Mindig egy documentation nevű branchre push-oláskor készüljön el? Egyszerűen kapcsoljuk ki az idő nagy részében az actiont?  

Előre köszönöm a válaszokat!

Reply:
wtf was I talking about? Pedig nem is ittam. (amugy az egerem kozepso gombja neha onallo eletre kel, az meg beszurja a legutoljara kijelolt dolgot, szoval ez tortent valoszinuleg es nem vettem eszre. sry about that.)

A modositasokat mindjart nezem. 
1 ok
2 en legalabb a protectedhez megirnam meg, mert ha valaki szarmaztat belole, azt meg ugye  eleri. 
3 en azt csinalnam, hogy "master" es "documentation" branchekre menjen le. Az esetek 90%-ban magan a deployolasi folyamaton nem valtoztattok, szoval azt nem kell tesztelni, branchen lefuttatni. Az rossz is lenne, ha a brancheken levo felkesz dolgokkal frissulne. A master a lenyeg.A document branchre meg ugy futtatnam, hogy egy gh-pages-2-re deployoljon tesztnek. Es akkor ha magan a doxygenen hegesztetek valamit akkor azt ott csinalnam.